ui/lib/replication/addon/controllers/replication-mode.js

```diff
@@ -3,36 +3,74 @@
  * SPDX-License-Identifier: BUSL-1.1
  */
 
-import { alias } from '@ember/object/computed';
 import { service } from '@ember/service';
 import Controller from '@ember/controller';
 import { task, timeout } from 'ember-concurrency';
 import { waitFor } from '@ember/test-waiters';
+import { action } from '@ember/object';
 
-export default Controller.extend({
-  router: service(),
-  rm: service('replication-mode'),
-  replicationMode: alias('rm.mode'),
-  waitForNewClusterToInit: task(
-    waitFor(function* (replicationMode) {
-      // waiting for the newly enabled cluster to init
-      // this ensures we don't hit a capabilities-self error, called in the model of the mode/index route
-      yield timeout(1000);
-      this.router.transitionTo('vault.cluster.replication.mode', replicationMode);
-    })
-  ),
-  actions: {
-    onEnable(replicationMode, mode) {
-      if (replicationMode == 'dr' && mode === 'secondary') {
+export default class ReplicationModeBaseController extends Controller {
+  @service('replication-mode') rm;
+  @service() router;
+  @service store;
+
+  get replicationMode() {
+    return this.rm.mode;
+  }
+
+  get replicationForMode() {
+    if (!this.replicationMode || !this.model) return null;
+    return this.model[this.replicationMode];
+  }
+
+  @task
+  @waitFor
+  *waitForNewClusterToInit(replicationMode) {
+    // waiting for the newly enabled cluster to init
+    // this ensures we don't hit a capabilities-self error, called in the model of the mode/index route
+    yield timeout(1000);
+    this.router.transitionTo('vault.cluster.replication.mode', replicationMode);
+  }
+
+  @action
+  onDisable() {
+    this.router.transitionTo('vault.cluster.replication.index');
+  }
+
+  @action
+  async onEnableSuccess(resp, replicationMode, clusterMode, doTransition = false) {
```

Comment:
Now that the enable form is a component - I'm curious why this is defined in the controller? Both functions are passed in the template the same (without args), it looks like the args are passed when the function is invoked in the component anyway

Reply:
Great question -- I have it on the parent because I wanted to avoid having to pass in the cluster model into the form component. Also, it felt like a side effect that is not intrinsic to the form itself, so it made sense to pass in from higher up

Reply:
That makes sense! Thanks for the explanation 😄